src/cpp/utils/SystemInfo.cpp

```diff
@@ -155,9 +155,18 @@ fastdds::dds::ReturnCode_t SystemInfo::get_username(
 bool SystemInfo::file_exists(
         const std::string& filename)
 {
+#ifdef _WIN32
+    // modify for mingw
+    DWORD fileAttributes = GetFileAttributesA(filename.c_str());
+    if (fileAttributes == INVALID_FILE_ATTRIBUTES) 
```

Comment:
And the last one to pass uncrustify CI
```suggestion
    if (fileAttributes == INVALID_FILE_ATTRIBUTES)
```